README.md
# NAME

Dist::Zilla::Plugin::PodnameFromClassname - Insert PODNAME for Moops classes.

<div>
    <p>
    <img src="https://img.shields.io/badge/perl-5.10+-blue.svg" alt="Requires Perl 5.10+" />
    <a href="https://travis-ci.org/Csson/p5-Dist-Zilla-Plugin-PodnameFromClassname"><img src="https://api.travis-ci.org/Csson/p5-Dist-Zilla-Plugin-PodnameFromClassname.svg?branch=master" alt="Travis status" /></a>
    <a href="http://cpants.cpanauthors.org/dist/Dist-Zilla-Plugin-PodnameFromClassname-0.1100"><img src="https://badgedepot.code301.com/badge/kwalitee/Dist-Zilla-Plugin-PodnameFromClassname/0.1100" alt="Distribution kwalitee" /></a>
    <a href="http://matrix.cpantesters.org/?dist=Dist-Zilla-Plugin-PodnameFromClassname%200.1100"><img src="https://badgedepot.code301.com/badge/cpantesters/Dist-Zilla-Plugin-PodnameFromClassname/0.1100" alt="CPAN Testers result" /></a>
    <img src="https://img.shields.io/badge/coverage-81.4%-orange.svg" alt="coverage 81.4%" />
    </p>
</div>

# VERSION

Version 0.1100, released 2016-04-02.

# SYNOPSIS

In dist.ini:

    [PodnameFromClassname]

In a [Moops](https://metacpan.org/pod/Moops) class:

    # PODCLASSNAME

    class My::Class { ... }

Results in:

    # PODNAME: My::Class

    class My::Class { ... }

# DESCRIPTION

Dist::Zilla::Plugin::PodnameFromClassname is useful together with [Moops](https://metacpan.org/pod/Moops) and [Pod::Weaver](https://metacpan.org/pod/Pod::Weaver). Since Moops classes generally don't also have a `package` statement `Pod::Weaver` can't pick up the module name.

Using this plugin and adding a `# PODCLASSNAME` line fixes that (by replacing such a line with the standard `# PODNAME: $classname` comment). There should be only one `# PODCLASSNAME` comment per file.

The plugin picks the first line that starts with `class`, `role`, `namespace` or `library` and uses the package name given on that line.

The comment line should match this regular expression:

    qr/^[\t ]*# PODCLASSNAME:?$/ms

Note: While this plugin was developed with [Moops](https://metacpan.org/pod/Moops) in mind, it should work with any module that introduces any of those keywords.

# SEE ALSO

- [Dist::Zilla::Plugin::PodnameFromFilename](https://metacpan.org/pod/Dist::Zilla::Plugin::PodnameFromFilename)
- [Dist::Zilla](https://metacpan.org/pod/Dist::Zilla)

# SOURCE

[https://github.com/Csson/p5-Dist-Zilla-Plugin-PodnameFromClassname](https://github.com/Csson/p5-Dist-Zilla-Plugin-PodnameFromClassname)

# HOMEPAGE

[https://metacpan.org/release/Dist-Zilla-Plugin-PodnameFromClassname](https://metacpan.org/release/Dist-Zilla-Plugin-PodnameFromClassname)

# AUTHOR

Erik Carlsson <info@code301.com>

# COPYRIGHT AND LICENSE

This software is copyright (c) 2016 by Erik Carlsson.

This is free software; you can redistribute it and/or modify it under
the same terms as the Perl 5 programming language system itself.


Changes
Revision history for Dist-Zilla-Plugin-PodnameFromClassname

{{$NEXT}}
 [API Changes]

 [Bug Fixes]

 [Documentation]

 [Enhancements]

 [New Features]

 [Other Changes]

 [Requirements]

 [Stats & Dependency Changes]

 [Testing]

0.1100 2016-04-02T20:47:03Z UTC
 [Bug Fixes]
 - $VERSION was outside the package declaration (fixes #1, thanks
   BAREFOOT).

 [Enhancements]
 - Removed the rules regarding empty lines around the PODCLASSNAME comment.

 [Stats & Dependency Changes]
 - 7 files changed, 103 insertions(+), 36 deletions(-)
 - (run req) + Dist::Zilla::Role::FileFinderUser 5.000
 - (run req) + Dist::Zilla::Role::FileMunger 5.000
 - (run req) + namespace::autoclean 0.22
 - (run req) ~ Moose (any) --> 2.1405
 - (run req) ~ perl 5.010001 --> 5.010000
 - (run req) - Dist::Zilla
 - (test req) + ExtUtils::MakeMaker (any)
 - (test req) + File::Spec (any)
 - (test req) + IO::Handle (any)
 - (test req) + IPC::Open3 (any)
 - (test req) + Path::Tiny 0.072
 - (test req) + Test::DZil 5.000
 - (test req) + Test::Differences 0.50
 - (test rec) + CPAN::Meta 2.120900
 - (conf req) + ExtUtils::MakeMaker (any)
 - (conf req) - Module::Build::Tiny
 - (dev req) + Badge::Depot 0.0103
 - (dev req) + Badge::Depot::Plugin::Coverage (any)
 - (dev req) + Badge::Depot::Plugin::Cpantesters (any)
 - (dev req) + Badge::Depot::Plugin::Kwalitee (any)
 - (dev req) + Badge::Depot::Plugin::Perl (any)
 - (dev req) + Badge::Depot::Plugin::Travis (any)
 - (dev req) + Dist::Zilla 5.000
 - (dev req) + Dist::Zilla::Plugin::Authority 1.009
 - (dev req) + Dist::Zilla::Plugin::BumpVersionAfterRelease::Transitional
   0.007
 - (dev req) + Dist::Zilla::Plugin::ChangeStats::Dependencies::Git 0.0100
 - (dev req) + Dist::Zilla::Plugin::CheckChangesHasContent 0.008
 - (dev req) + Dist::Zilla::Plugin::Clean 0.07
 - (dev req) + Dist::Zilla::Plugin::ConfirmRelease (any)
 - (dev req) + Dist::Zilla::Plugin::CopyFilesFromBuild 0.151680
 - (dev req) + Dist::Zilla::Plugin::DistIller::MetaGeneratedBy (any)
 - (dev req) + Dist::Zilla::Plugin::ExecDir (any)
 - (dev req) + Dist::Zilla::Plugin::Git 2.036
 - (dev req) + Dist::Zilla::Plugin::Git::Check (any)
 - (dev req) + Dist::Zilla::Plugin::Git::CheckFor::CorrectBranch 0.013
 - (dev req) + Dist::Zilla::Plugin::Git::Commit (any)
 - (dev req) + Dist::Zilla::Plugin::Git::Contributors 0.017
 - (dev req) + Dist::Zilla::Plugin::Git::GatherDir (any)
 - (dev req) + Dist::Zilla::Plugin::Git::Push (any)
 - (dev req) + Dist::Zilla::Plugin::Git::Tag (any)
 - (dev req) + Dist::Zilla::Plugin::GithubMeta 0.54
 - (dev req) + Dist::Zilla::Plugin::InstallRelease 0.008
 - (dev req) + Dist::Zilla::Plugin::License (any)
 - (dev req) + Dist::Zilla::Plugin::MakeMaker (any)
 - (dev req) + Dist::Zilla::Plugin::Manifest (any)
 - (dev req) + Dist::Zilla::Plugin::ManifestSkip (any)
 - (dev req) + Dist::Zilla::Plugin::MetaJSON (any)
 - (dev req) + Dist::Zilla::Plugin::MetaNoIndex (any)
 - (dev req) + Dist::Zilla::Plugin::MetaProvides::Package 2.003001
 - (dev req) + Dist::Zilla::Plugin::MetaYAML (any)
 - (dev req) + Dist::Zilla::Plugin::NextRelease::Grouped 0.0200
 - (dev req) + Dist::Zilla::Plugin::PodSyntaxTests (any)
 - (dev req) + Dist::Zilla::Plugin::PodWeaver 4.006
 - (dev req) + Dist::Zilla::Plugin::Prereqs (any)
 - (dev req) + Dist::Zilla::Plugin::Readme (any)
 - (dev req) + Dist::Zilla::Plugin::ReadmeAnyFromPod 0.150250
 - (dev req) + Dist::Zilla::Plugin::RewriteVersion::Transitional 0.007
 - (dev req) + Dist::Zilla::Plugin::RunExtraTests 0.028
 - (dev req) + Dist::Zilla::Plugin::ShareDir (any)
 - (dev req) + Dist::Zilla::Plugin::Test::Compile 2.054
 - (dev req) + Dist::Zilla::Plugin::Test::EOF 0.0501
 - (dev req) + Dist::Zilla::Plugin::Test::EOL 0.18
 - (dev req) + Dist::Zilla::Plugin::Test::Kwalitee::Extra v0.2.1
 - (dev req) + Dist::Zilla::Plugin::Test::NoTabs 0.15
 - (dev req) + Dist::Zilla::Plugin::Test::ReportPrereqs 0.021
 - (dev req) + Dist::Zilla::Plugin::Test::Version 1.05
 - (dev req) + Dist::Zilla::Plugin::TestRelease (any)
 - (dev req) + Dist::Zilla::Plugin::TravisYML 1.15
 - (dev req) + Dist::Zilla::Plugin::UploadToCPAN (any)
 - (dev req) + Pod::Elemental::Transformer::List (any)
 - (dev req) + Pod::Weaver::Plugin::SingleEncoding (any)
 - (dev req) + Pod::Weaver::Plugin::Transformer (any)
 - (dev req) + Pod::Weaver::PluginBundle::CorePrep (any)
 - (dev req) + Pod::Weaver::Section::Authors (any)
 - (dev req) + Pod::Weaver::Section::Badges 0.0401
 - (dev req) + Pod::Weaver::Section::Collect (any)
 - (dev req) + Pod::Weaver::Section::GenerateSection 1.01
 - (dev req) + Pod::Weaver::Section::Generic (any)
 - (dev req) + Pod::Weaver::Section::Leftovers (any)
 - (dev req) + Pod::Weaver::Section::Legal (any)
 - (dev req) + Pod::Weaver::Section::Name (any)
 - (dev req) + Pod::Weaver::Section::Region (any)
 - (dev req) + Pod::Weaver::Section::Version (any)
 - (dev req) + Test::EOF (any)
 - (dev req) + Test::EOL (any)
 - (dev req) + Test::More 0.88
 - (dev req) + Test::NoTabs 1.4
 - (dev req) + Test::Version 1
 - (dev req) + Test::Warnings 0.026
 - (dev sug) + Dist::Iller 0.1408
 - (dev sug) + Dist::Iller::Config::Author::CSSON 0.0309

0.1002 2015-01-15T12:38:19Z CET
 [Documentation]
 - Minor fixes and improvements.

0.1001 2015-01-15T04:31:55Z CET
 [Documentation]
 - Apparently, having [PodnameFromClassname] in dist.ini messed with the
   synopsis..

0.1000 2015-01-15T04:27:26Z CET
 - Initial release


Makefile.PL
# This file was automatically generated by Dist::Zilla::Plugin::MakeMaker v5.043.
use strict;
use warnings;

use 5.010000;

use ExtUtils::MakeMaker;

my %WriteMakefileArgs = (
  "ABSTRACT" => "Insert PODNAME for Moops classes.",
  "AUTHOR" => "Erik Carlsson <info\@code301.com>",
  "CONFIGURE_REQUIRES" => {
    "ExtUtils::MakeMaker" => 0
  },
  "DISTNAME" => "Dist-Zilla-Plugin-PodnameFromClassname",
  "LICENSE" => "perl",
  "MIN_PERL_VERSION" => "5.010000",
  "NAME" => "Dist::Zilla::Plugin::PodnameFromClassname",
  "PREREQ_PM" => {
    "Dist::Zilla::Role::FileFinderUser" => "5.000",
    "Dist::Zilla::Role::FileMunger" => "5.000",
    "Moose" => "2.1405",
    "namespace::autoclean" => "0.22"
  },
  "TEST_REQUIRES" => {
    "ExtUtils::MakeMaker" => 0,
    "File::Spec" => 0,
    "IO::Handle" => 0,
    "IPC::Open3" => 0,
    "Path::Tiny" => "0.072",
    "Test::DZil" => "5.000",
    "Test::Differences" => "0.50",
    "Test::More" => "0.96"
  },
  "VERSION" => "0.1101",
  "test" => {
    "TESTS" => "t/*.t"
  }
);


my %FallbackPrereqs = (
  "Dist::Zilla::Role::FileFinderUser" => "5.000",
  "Dist::Zilla::Role::FileMunger" => "5.000",
  "ExtUtils::MakeMaker" => 0,
  "File::Spec" => 0,
  "IO::Handle" => 0,
  "IPC::Open3" => 0,
  "Moose" => "2.1405",
  "Path::Tiny" => "0.072",
  "Test::DZil" => "5.000",
  "Test::Differences" => "0.50",
  "Test::More" => "0.96",
  "namespace::autoclean" => "0.22"
);


unless ( eval { ExtUtils::MakeMaker->VERSION(6.63_03) } ) {
  delete $WriteMakefileArgs{TEST_REQUIRES};
  delete $WriteMakefileArgs{BUILD_REQUIRES};
  $WriteMakefileArgs{PREREQ_PM} = \%FallbackPrereqs;
}

delete $WriteMakefileArgs{CONFIGURE_REQUIRES}
  unless eval { ExtUtils::MakeMaker->VERSION(6.52) };

WriteMakefile(%WriteMakefileArgs);
